### ISSUE TITLE
Add the ability to only deploy cloudformation template

### ISSUE THREAD

**Issue by alexnederlof** [serverless/serverless#7988]
### Use case description

When using a compiled language, like in my case Typescript, it takes a good minute or so to compile. Even though there are no code changes, it then create bundle, upload it, and only then start on the cloud formation template.

Like most humans (I guess), I make mistakes in Cloudformation YAML files, so I want to be able to iterate on them pretty rapidly, and not wait for my typescript to compile and upload on every run. 

### Proposed solution

Add a `--cloud-config-only` flag to the deploy option, to just upload the cloudformation file.

**Comment by medikoo** [comment]
@alexnederlof you can publish prebuilt artifacts (cloudformation template) with `--package` option

**Comment by alexnederlof** [comment]
I thought so as well, but here's the problem:

1. Starting with a clean slate, I first do `sls deploy`. Everything uploads correctly.
1. Then I make changes to the cloudformation file, and hit `sls deploy --package .serverless`. Everything seems to work, but the changes are not uploaded to cloudfront. The old cloudfront file is, without my changes.
1. Further more, now the `.serverless` folder has been removed too. 

In case you suspect this is a plugin problem, I'm using:

```
plugins:
    - serverless-plugin-scripts
    - serverless-plugin-typescript
    - serverless-offline
```

**Comment by medikoo** [comment]
> Then I make changes to the cloudformation file

Do you mean you're editing CF file as generated by the Framework?

> In case you suspect this is a plugin problem, I'm using:

If it's a plugin problem, it should be reported at plugin repository. Best if you ensure you have some problem when not relying on any plugins

**Comment by alexnederlof** [comment]
> Do you mean you're editing CF file as generated by the Framework?

Yes, this happens when I change something in `serverless.yaml`.

**Comment by medikoo** [comment]
> Yes, this happens when I change something in serverless.yaml.

Ok, so you meant that you've made changes to `serlveress.yml` not to cloudformation file, right ?

**Comment by alexnederlof** [comment]
Yes, my bad I should have been more clear, I've updated the description!

**Comment by medikoo** [comment]
With `--package` option you point that it's not what you have in `serverless.yml` that should be deployed, but artifacts that are at path you point by`--package` param.

With serverless you either can (1) Build and deploy current state (2) Deploy previously built state (pointed by `--package` param). What exactly are you after?

**Comment by alexnederlof** [comment]
I would like serverless to have a `(cloud?-)configuration-only` parameter, that assumes the code has not change, and just regenerates the cloudformation file and applies that. 

The reason is right now, the long iteration cycle of build - upload - wait for cloudformation to apply - see error - start over, takes so long it's frustrating. We end up extracting the cloudformation from the `.serverless` directory, converting that to YAML (because working in JSON is not nicey), then modifying that until it works, and then kind of merging it back into serverless.

**Comment by medikoo** [comment]
> I would like serverless to have a (cloud?-)configuration-only parameter, that assumes the code has not changed, and just regenerates the cloudformation file and applies that.

@alexnederlof There are few issues that I think makes it this feature not really sensible on Framework core grounds

Firstly Framework operates on files which are as is ready to deploy (if you use TypeScript, it's typescript plugin that should first prepare files, and it's after that Framework jumps in - technically Framework is not aware you're transpiling).

While I believe real bottle neck on your side it's this compilation time, and that is not in a Framework scope (request for eventual optimization on that should be directed to repo of plugin that handles transpilation).

Having that clarified, question that remains, is it makes sense to optimize deployment of files, so we generate package and upload only if it appaers different from one uploaded. It could save some bits of time, depending on connection, but I think it won't be noticeable in context of whole deployment time, where it's CloudFormation deployment that's the real bottleneck.
Therefore I take feature is not worth investing time into it (and more importantly addressing that won't solve the issue on your side, where it's transpilation time that's real bottleneck)

**Comment by alexnederlof** [comment]
I see your point, but I would argue that in even without transpiling this would be worth it. I also work with serverless for python, where I also am often waiting for unchanged code to be deployed, before getting feedback on my cloud config. 

On Python, without making any code changes, I go through the same loop but here the "excluding development dependencies" is slower, it has to compile via docker to get the right deps in (because of how python dependencies on different platforms work), and the upload is much slower (because of the size). 

Now obviously, one could argue that that's the fault of the python-dependency-plugin, just like the typscript plugin. But I would assume a lot of people use these languages (and setup) and would greatly benefit from this suggestion.

**Comment by medikoo** [comment]
> I would argue that in even without transpiling this would be worth it. I also work with serverless for python, where I also am often waiting for unchanged code to be deployed, before getting feedback on my cloud config.

When speaking of full service deployment, I'm pretty sure the bottlenecks are: 
1. CloudFormation deployment time (but it has to be measured whether not updating artifacts will have significant impact on that, if its the case, then indeed it might be worth pursuing)
2. Packaging time, of which design is actually broken in Framework itself, and that issue is handled at #6580 

I would keep it on hold until #6580 is addressed. After it's addressed we may do some real world measurements and confirm whether benefit will indeed be observable.

**Comment by alexnederlof** [comment]
Ok, thanks for thinking along @medikoo, and your timely replies. Much appreciated! 😄

**Comment by spullara** [comment]
I thought there was a claim in the documentation that if the code didn't change it won't upload new code but I think that only applies if nothing at all changes. I would like it to not update function code whose .zip files didn't change.

**Comment by medikoo** [comment]
There's another related issue on how to optimize deployments. by reusing exactly same artifacts that were previously uploaded: https://github.com/serverless/serverless/issues/8666
